### ISSUE TITLE
TLSUtil ParseSigner Support RSA

### ISSUE THREAD

**Issue by FriedCircuits** [hashicorp/consul#7622]
#### Feature Description

Add support in the TLS Util for RSA private key to generate certs for consul from our existing CA so its trusted already. 

```bash
consul tls cert create -server -days=730 -ca ca.crt -key ca.key  -domain consul
```

https://github.com/hashicorp/consul/blob/9ea1a204d832cfd4d8c5cba9df56876a526a4531/tlsutil/generate.go#L166-L179

called from 

https://github.com/hashicorp/consul/blob/bf6a91af9454c90273fa3039798c787f7f61aaac/command/tls/cert/create/tls_cert_create.go#L173

#### Use Case(s)

Long story short I had an issue with "unknown PEM block type for signing key When helm would run TLS init. I traced it down to the consul TLS create cert command and ran that myself. That lead me to the function ParseSinger which in TLSUtil looks to only support EC. I would like to be able to use our RSA key instead of letting Conul generate a key. Is possible to add support?

**Comment by FriedCircuits** [comment]
The connect agent seems to support it unless there is more to it?

https://github.com/hashicorp/consul/blob/7f039494247f56eea1f0eebd7f911bf0bbecb8c9/agent/connect/parsing.go#L112

**Comment by FriedCircuits** [comment]
Any change for a review on this? I would like to get this solved for our testing env so we can roll it out. Thanks.

**Comment by mfuxi** [comment]
I'm having the same issue as well, is there any chance to fix it?
TL;DR - it's impossible to use RSA certs with the Consul helm chart

**Comment by jsosulska** [comment]
This should have been closed with #7623. Closing now.

**Comment by tjhiggins** [comment]
@jsosulska can you please add a backport label on #7623? It didn't make it into 1.9.4.

**Comment by jsosulska** [comment]
Hi @tjhiggins ,

Thanks for following up! For enhancements like this, the Consul team will hold it to release until the next minor release (e.g. 1.10). I'll post here when we have a RC/Beta ready to drop, as we would love additional feedback on the release and other enhancements too :)

Best,
Jono
